src/utils/fs.rs

```diff
@@ -80,6 +80,77 @@ pub fn nice_directory_display(os_str: impl AsRef<OsStr>) -> Cow<'static, str> {
     }
 }
 
+/// Try to detect the file extension by looking for known magic strings
+pub fn try_infer(path: &Path) -> Option<Extension> {
+    fn is_zip(buf: &[u8]) -> bool {
+        buf.len() > 3
+            && buf[0] == 0x50
+            && buf[1] == 0x4B
+            && (buf[2] == 0x3 || buf[2] == 0x5 || buf[2] == 0x7)
+            && (buf[3] == 0x4 || buf[3] == 0x6 || buf[3] == 0x8)
+    }
+    fn is_tar(buf: &[u8]) -> bool {
+        buf.len() > 261
+            && buf[257] == 0x75
+            && buf[258] == 0x73
+            && buf[259] == 0x74
+            && buf[260] == 0x61
+            && buf[261] == 0x72
+    }
+    fn is_gz(buf: &[u8]) -> bool {
+        buf.len() > 2 && buf[0] == 0x1F && buf[1] == 0x8B && buf[2] == 0x8
+    }
+    fn is_bz2(buf: &[u8]) -> bool {
+        buf.len() > 2 && buf[0] == 0x42 && buf[1] == 0x5A && buf[2] == 0x68
+    }
+    fn is_xz(buf: &[u8]) -> bool {
+        buf.len() > 5
+            && buf[0] == 0xFD
+            && buf[1] == 0x37
+            && buf[2] == 0x7A
+            && buf[3] == 0x58
+            && buf[4] == 0x5A
+            && buf[5] == 0x00
+    }
+    fn is_lz(buf: &[u8]) -> bool {
+        buf.len() > 3 && buf[0] == 0x4C && buf[1] == 0x5A && buf[2] == 0x49 && buf[3] == 0x50
+    }
+    fn is_lz4(buf: &[u8]) -> bool {
+        buf.len() > 3 && buf[0] == 0x04 && buf[1] == 0x22 && buf[2] == 0x4D && buf[3] == 0x18
+    }
+    fn is_zst(buf: &[u8]) -> bool {
+        buf.len() > 3 && buf[0] == 0x28 && buf[1] == 0xB5 && buf[2] == 0x2F && buf[3] == 0xFD
+    }
+
+    let buf = {
+        let mut b = [0; 270];
+        // Reading errors will just make the inferring fail
+        let _rn = std::fs::File::open(&path).map(|mut f| std::io::Read::read(&mut f, &mut b));
```

Comment:
What `_rn` means?

Reply:
I can just change it to _ ( I think in my mind was something like result number)